packages/payload/src/admin/components/elements/ListDrawer/DrawerContent.tsx

```diff
@@ -198,6 +198,10 @@ export const ListDrawerContent: React.FC<ListDrawerProps> = ({
     setPreference(preferenceKey, newPreferences, true)
   }, [sort, limit, setPreference, preferenceKey])
 
+  useEffect(() => {
```

Comment:
Hmmm, this isn't quite right. Instead of adding a new affect, you should be able to tie into the methods that the `SearchFilter` component exposed via props. The drawer sets `modifySearchParams` to `false`, which is where the page reset is being bypassed. Basically just need to thread the page reset back into the search _before_ it's set, instead of _after_ its set.

Reply:
Notes:
- Consolidate all 5 useState s in the ListDrawer DrawerContent component
- Replace them with a new params state, and completely remove the useEffect
- This would mean that the filter options injected into the where query, and the conditions formatted from the search string, need to become methods that properly set params
- This is exactly what a reducer is for, might as well use it. We can create reducer actions like setSearch which properly formats conditions and page , and setWhere, setParams to blindy set any of them, etc.
- This all makes it so that the usePayloadAPI hook (which makes an API request) is only fired once no matter what params change and when